tests/test_saturation.py

```diff
@@ -76,7 +75,8 @@ def test_read_pattern_saturation_flagging():
     )
 
     # Make sure that groups after the third get flagged
-    assert np.all(gdq[0, 2:, 5, 5] == DQFLAGS["SATURATED"])
+    assert np.all(gdq[0, 3:, 5, 5] == DQFLAGS["SATURATED"])
```

Comment:
I think the behavior introduced by #321 is not desirable long term; if we opt to remove the xfail here, we should at least add a comment describing the ideal state for future edits to the code and this test. I believe the long-term fix will be tracked under https://jira.stsci.edu/browse/JP-3835